lib/util.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2018-2019 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const {util: {binary: {base58}}} = require('node-forge');
+
+/**
+ * Wraps Base58 decoding operations in
+ * order to provide consistent error messages.
+ * @ignore
+ * @example
+ * > const pubkeyBytes = _base58Decode({
+ *    decode: base58.decode,
+ *    keyMaterial: this.publicKeyBase58,
+ *    type: 'public'
+ *   });
+ * @param {Object} options - The decoder options.
+ * @param {Function} options.decode - The decode function to use.
+ * @param {string} options.keyMaterial - The Base58 encoded
+ * key material to decode.
+ * @param {string} options.type - A description of the
+ * key material that will be included
+ * in an error message (e.g. 'public', 'private').
+ *
+ * @returns {Object} - The decoded bytes. The data structure for the bytes is
+ *   determined by the provided decode function.
+ */
+exports.base58Decode = ({decode, keyMaterial, type}) => {
```

Comment:
I get that we have legacy code that uses these anonymous arrow functions. But for new code, would it be possible to use named functions in these cases?